### PR TITLE
try to backport the named parameter patch

### PR DESCRIPTION
**Pull request by pillowtrucker**

### CODE REVIEW

File: inox2d/src/params.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ fn ranges_out(
 #[derive(Debug, Clone)]
 pub struct Param {
 	pub uuid: u32,
-	pub name: String,
+	pub name: Option<String>,
```
Comment (Speykious) [inline inox2d/src/params.rs:53]: Huh. A parameter may not have a name? That's so weird.
But I guess I can see it happen as long as there's a UUID :v

File: inox2d/Cargo.toml [hunk 2]
```diff
@@ -22,3 +22,9 @@ clap = { version = "4.1.8", features = ["derive"] }
 
 [features]
 owo = ["dep:owo-colors"]
+[dependencies.uuid]
+version = "1.7"
+features = [
+    "v4",                # Lets you generate random UUIDs
+    "fast-rng",          # Use a faster (but still sufficiently random) RNG
+]
```
Comment (Speykious) [inline inox2d/Cargo.toml]: What's the benefit of introducing `uuid` as a dependency here? We don't need to generate UUIDs at any point, I don't see the necessity.

**Comment by pillowtrucker** [comment]
I'm still not getting the whole arrows example for physics rendered even with this patch..
First run is with this patch, second main branch, third physics branch: https://dai.ly/k2FyniuCiLtKjCzXsFk

File: inox2d/src/params.rs [hunk 3]
```diff
@@ -191,8 +193,9 @@ impl Puppet {
 
 	pub fn set_param(&mut self, param_name: &str, val: Vec2) {
 		let param = self
-			.parameters
+			.param_names
 			.get(param_name)
+			.map(|pn| self.parameters.get(pn).expect("param index mismatch {pn}"))
 			.unwrap_or_else(|| panic!("No parameter named: {}", param_name));
```
Comment (Speykious) [inline inox2d/src/params.rs:205]: `.expect` is not a format-like macro, it's just a function. So that `{pn}` is gonna display as `{pn}` and not the contents of `pn`. I suggest either removing that part, or (better solution) make `get/set_param` return a `Result<T, ParamNotFound>`, which will automatically also make the methods more resilient like we wanted a few days ago

File: test/test-physics/Cargo.toml
Comment (Speykious) [inline test/test-physics/Cargo.toml:1]: This crate is basically a clone of the `render-wgpu` example, I'd prefer not pulling that in

**Review by Speykious — CHANGES_REQUESTED** [review]
I didn't know parameters could be unnamed, it's good to use the provided UUIDs as IDs instead when we can. However, *I requesteth changes*

**Comment by Speykious** [comment]
Just realized parameters already have use the UUID in the `physics` branch...

File: inox2d/Cargo.toml (hunk 2 above)
Comment (pillowtrucker) [reply inox2d/Cargo.toml]: they're cheap to clone, actually 128-bit uuids, and bring free serialization. I would suggest at least changing the existing implementation to take 64-bit unsigned integers instead of u32, as that's what most javascript runtime will actually produce as a Number

File: inox2d/src/params.rs (hunk 3 above)
Comment (pillowtrucker) [reply inox2d/src/params.rs:205]: Ehh yeah I wasn't paying attention to this so it might already be a Result in the other branch ? I'll check and copy
The inner error is a panic error though, the index having a reference to a non-existent value

File: test/test-physics/Cargo.toml
Comment (pillowtrucker) [reply test/test-physics/Cargo.toml:1]: Any idea what else to do so that we don't keep changing those 3 lines in all 3 examples each time we switch branch and want to test a different model with different params ?

**Comment by pillowtrucker** [comment]
TLDR; I'm not really convinced this should be merged but I'm also not convinced the physics and main branch haven't diverged so much that a rebase will be just as much work as a rewrite

File: test/test-physics/Cargo.toml
Comment (Speykious) [reply test/test-physics/Cargo.toml:1]: Maybe have the parameter we want to test be optionally passed in as a CLI argument using `clap`

File: inox2d/Cargo.toml (hunk 2 above)
Comment (Speykious) [reply inox2d/Cargo.toml]: Currently [the reference implementation has `Param.uid` as a `uint`](https://github.com/Inochi2D/inochi2d/blob/2d1a8014a1a7aa352d40a26773c34107cfb7394e/source/inochi2d/core/param/package.d#L143-L146), which in D is a signed 32-bit integer. If inochi2d gets eventually patched to use an actual UUID as a parameter ID, I'll gladly do this, but I think it's best to keep the parameter ID a `u32` for now.

**Comment by Speykious** [comment]
Physics is now fixed on the physics branch, so I think I'll close this
